sklearn/feature_selection/_rfe.py

```diff
@@ -694,7 +777,7 @@ def __init__(
         # RFECV.estimator is not validated yet
         prefer_skip_nested_validation=False
     )
-    def fit(self, X, y, groups=None):
+    def fit(self, X, y, groups=None, **params):
```

Comment:
we should make `groups` kwonly here, and then later remove it from the signature. This should also be removed from requests on the `set_fit_request` of this estimator, which can be done as:

```py
    __metadata_request__fit = {"groups": metadata_routing.UNUSED}
```
set as a class attribute here.